engine/apps/alerts/models/alert_group.py

```diff
@@ -715,38 +711,6 @@ def resolve_by_source(self):
         for dependent_alert_group in self.dependent_alert_groups.all():
             dependent_alert_group.resolve_by_source()
 
-    # deprecated
-    def resolve_by_archivation(self):
-        AlertGroupLogRecord = apps.get_model("alerts", "AlertGroupLogRecord")
-        # if incident was silenced, unsilence it without starting escalation
-        if self.silenced:
-            self.un_silence()
-            self.log_records.create(
-                type=AlertGroupLogRecord.TYPE_UN_SILENCE,
-                silence_delay=None,
-                reason="Resolve by archivation",
-            )
-        self.archive()
-        self.stop_escalation()
-        if not self.resolved:
-            self.resolve(resolved_by=AlertGroup.ARCHIVED)
-
-            log_record = self.log_records.create(type=AlertGroupLogRecord.TYPE_RESOLVED)
-
-            logger.debug(
-                f"send alert_group_action_triggered_signal for alert_group {self.pk}, "
-                f"log record {log_record.pk} with type '{log_record.get_type_display()}', action source: archivation"
-            )
-
-            alert_group_action_triggered_signal.send(
-                sender=self.resolve_by_archivation,
-                log_record=log_record.pk,
-                action_source=None,
-            )
-
-        for dependent_alert_group in self.dependent_alert_groups.all():
-            dependent_alert_group.resolve_by_archivation()
-
```

Comment:
`resolve_by_archivation` was only ever invoked in the following two spots:

- `CheckAlertIsUnarchivedMixin.check_alert_is_unarchived`: this method seemed to be deprecated because the logic inside the method was as such:
```python3
def check_alert_is_unarchived(self, slack_team_identity, payload, alert_group, warning=True):
    alert_group_is_unarchived = alert_group.started_at.date() > self.organization.archive_alerts_from
    if not alert_group_is_unarchived:
        if warning:
            warning_text = "Action is impossible: the Alert is archived."
            self.open_warning_window(payload, warning_text)
        if not alert_group.resolved or not alert_group.is_archived:
            alert_group.resolve_by_archivation()
    return alert_group_is_unarchived
```

`organization.archive_alerts_from` is **always** set to `1970-01-01` (only value that exists in our production database) so therefore the above logic will _never_ invoke `AlertGroup.resolve_by_archivation`

- in the `resolve_archived_incidents_for_organization` Celery task

this celery task was only ever queued by `CurrentOrganizationSerializer.update`:
```python3
def update(self, instance, validated_data):
    current_archive_date = instance.archive_alerts_from
    archive_alerts_from = validated_data.get("archive_alerts_from")

    result = super().update(instance, validated_data)
    if archive_alerts_from is not None and current_archive_date != archive_alerts_from:
        if current_archive_date > archive_alerts_from:
            unarchive_incidents_for_organization.apply_async(
                (instance.pk,),
            )
        resolve_archived_incidents_for_organization.apply_async(
            (instance.pk,),
        )

    return result
```

`archive_alerts_from` will always be `None` here